### ISSUE TITLE
CRAN issue with not selecting the correct C++ compiler in configure tests

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#766]
Email from CRAN:

This concerns packages

RProtoBuf RcppRedis SOD divest freetypeharfbuzz imager nimble nloptr polyclip rgdal zstdr

All show the line

checking whether g++ accepts -g... yes

in configure output on platforms where the C++ compiler is clang++ . As 'Writing R Extensions' §1.2 says

<quote>
packages written with C++ need to pick up the details for the C++ compiler and switch the current language to C++ by something like

CXX=`"${R_HOME}/bin/R" CMD config CXX`
CXXFLAGS=`"${R_HOME}/bin/R" CMD config CXXFLAGS`
AC_LANG(C++)
</quote>

where the CXX* variables should be set before a call to AC_LANG(C++) or AC_PROG_CXX, and both are needed.  (Setting them later probably works but is wasteful. If you compile in a subdirectory the variables need to be passed down to sub-makes.)

This may or may not matter depending on what configure tests for -- it was recently discovered to matter for OpenMP in a couple of packages, and also matters when using a 32-bit build on x86_64 Linux and on systems with compiler-specific includes/libraries (e.g. for clang++ vs g++).

Please correct at the next release.

Followup email from Dirk:

To be safe it should check whethe R_HOME is set. I have been using this for
years following a suggestion from Kurt a long time ago:

```
: ${R_HOME=`R RHOME`}
if test -z "${R_HOME}"; then
    AC_MSG_ERROR([Could not determine R_HOME.])   
fi
```
which should then precede the use of R_HOME in

**Comment by paciorek** [comment]
Just realized that we had the stanza that CRAN is requesting but we commented it out in favor of:

```
CXX11=`"${R_HOME}/bin/R" CMD config CXX1X`
CXX11STD=`"${R_HOME}/bin/R" CMD config CXX1XSTD`
CXX="${CXX11} ${CXX11STD}"
CXXFLAGS=`"${R_HOME}/bin/R" CMD config CXX1XFLAGS`
AC_LANG(C++)
```

**Comment by paciorek** [comment]
I'm going to dig further later, but @perrydv if you recall when we added the CXX11 support and have any thoughts here, please weigh in.

**Comment by paciorek** [comment]
Possible solution:
move AC_PROG_CXX  in configure.ac to after we define CXX, CXX1X etc.

Still not sure if we should change "1X" to "11". On recent versions of R, CXX11 seems to be defined in etc/Makeconf (CXX1X is still defined after a #legacy comment), but in an old R 3.3.3, only CXX1X is defined in etc/Makeconf. So concern would be backwards compatibility.

**Comment by paciorek** [comment]
Fixes PR #767
